app/app.go

```diff
@@ -1038,19 +1038,51 @@ func (app *App) LastLogs(lines int, filterLog Applog) ([]Applog, error) {
 	return logs, nil
 }
 
+type Filter struct {
+	Name      *regexp.Regexp
+	Platform  string
+	TeamOwner string
+	UserOwner string
+}
+
```

Comment:
hi @andrewsmedina .. yes, the solution that I've found is working, the tests are passing but its not as clean as the solution @fsouza suggested:

```
func (f *Filter) Query() bson.M {
    return bson.M{
        "name":      bson.M{"$regexp": f.Name.String()},
        "platform":  f.Platform,
        "teamowner": f.TeamOwner,
        "owner":     f.UserOwner,
    }
}
```

So if you have any suggestions to improve my solution , please let me know :)